### PR TITLE
Added InversifyJS type definitions

### PR DESCRIPTION
**Pull request by remojansen**
Added InversifyJS type definitions. Includes definitions for the following npm modules:
- inversify
- inversify-binding-decorators  
- inversify-express-utils
- inversify-logger-middleware

### CODE REVIEW

File: inversify/inversify.d.ts [hunk 1]
```diff
@@ -1,141 +1,273 @@
...
-    export var named: any;
-    export var paramNames: any;
+    export var Kernel: interfaces.KernelConstructor;
+    export var KernelModule: interfaces.KernelModuleConstructor;
+    export var decorate: (decorator: (ClassDecorator|ParameterDecorator), target: any, parameterIndex?: number) => void;
+    export function injectable(): (typeConstructor: any) => void;
+    export function tagged(metadataKey: string, metadataValue: any): (target: any, targetKey: string, index?: number) => any;
+    export function named(name: string): (target: any, targetKey: string, index?: number) => any;
+    export function targetName(name: string): (target: any, targetKey: string, index: number) => any;
+    export function inject(serviceIdentifier: (string|Symbol|any)): (target: any, targetKey: string, index?: number) => any;
```
Comment (mhegazy) [inline inversify/inversify.d.ts]: this does not look right. `any|...` is just `any`. there is no value of adding the two other types.

Comment (mhegazy) [inline inversify/inversify.d.ts]: consider changing this to `{}` instead.

Comment (mhegazy) [inline inversify/inversify.d.ts]: there are multiple references for this throughout the PR.

**Comment by remojansen** [comment]
Hi @mhegazy thanks for the code review. I have replaced:
```ts
(string|Symbol|any)
```
With:
```
ServiceIdentifier<any>
```
The ServiceIdentifier is:
```
export type ServiceIdentifier<T> = (string|Symbol|Newable<T>);
```

**Comment by mhegazy** [comment]
thanks!

**Pull request merged**